connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -892,6 +893,9 @@ message MapPartitions {
 
   // (Optional) Whether to use barrier mode execution or not.
   optional bool is_barrier = 3;
+
+  // (Optional) ResourceProfile id used for the stage level scheduling.
+  optional int32 profile_id = 4;
```

Comment:
Shouldn't these be uuids? Just to make sure that we have intentional difference and no off by one?

Reply:
I'm not quite following you about "by uuids".

So the basic implementation is

1. The client creates ResourceProfile
2. if the profile ID of ResourceProfile is accessed for the first time, then the client will ask to create ResourceProfile and add it to the ResourceProfileManager on the server side, and the server side will return the profile ID to the client which will set the id to the ResourceProfile on the client side.
3. The internal mapInPandas/mapInArrow will just use the ResourceProfile id, and the server side can extract the ResourceProfile from ResourceProfileManager according to the id.